### PR TITLE
fix italic and small font for "alt hint" in chart tooltip

### PR DESCRIPTION
**Pull request by mierin12**
Closes https://github.com/portfolio-performance/portfolio/issues/4318

Hello, I do not know why, but this little change works on my Windows 11.
**Before :** 
2024-11-18 20_27_30-Portfolio Performance
 
**After :**
2024-11-18 20_21_25-Portfolio Performance

it seems to be thanks to `hint.getParent().pack();` recently added. When `setFont` is called after it, it works for me.

### CODE REVIEW

**Comment by Nirus2000** [comment]
Hello @mierin12 
This also works for me under Windows10. Very good. 👍🏻 
grafik


---

However, I do have one comment. I have noticed that the translations are different.

**I have the following text in German translation:**
_'Alt' ('Option' unter macOS) beim Öffnen des Tooltips sortiert die Einträge nach Wert_

As far as I know, there should be a function for recognizing which operating system is being used.

**Comment by Nirus2000** [comment]
Perfect... 😉 👍🏻

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/TimelineChartToolTip.java [hunk 1]
```diff
@@ -227,7 +228,10 @@ protected void createComposite(Composite parent)
         extraInfoProvider.forEach(provider -> provider.accept(container, focus));
 
         Label hint = new Label(data, SWT.WRAP);
-        hint.setText(Messages.TooltipHintPressAlt);
+        String os = System.getProperty("os.name").toLowerCase(); //$NON-NLS-1$
+        String pressAltOrOption = os.startsWith("mac") ? "Option" : "Alt"; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+        hint.setText(MessageFormat.format(Messages.TooltipHintPressAlt, pressAltOrOption));
```
Comment (Nirus2000) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/TimelineChartToolTip.java:233]: hint.setText(MessageFormat.format(Messages.TooltipHintPressAlt, 
    Platform.OS_MACOSX.equals(Platform.getOS()) ? "Option" : "Alt"));

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/TimelineChartToolTip.java [hunk 2]
```diff
@@ -227,16 +228,19 @@ protected void createComposite(Composite parent)
         extraInfoProvider.forEach(provider -> provider.accept(container, focus));
 
         Label hint = new Label(data, SWT.WRAP);
-        hint.setText(Messages.TooltipHintPressAlt);
-        hint.setFont(this.resourceManager
-                        .create(FontDescriptor.createFrom(data.getFont()).increaseHeight(-3).withStyle(SWT.ITALIC)));
+        String os = System.getProperty("os.name").toLowerCase(); //$NON-NLS-1$
+        String pressAltOrOption = os.startsWith("mac") ? "Option" : "Alt"; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+        hint.setText(MessageFormat.format(Messages.TooltipHintPressAlt, pressAltOrOption));
+
```
Comment (Nirus2000) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/TimelineChartToolTip.java:234]:
```suggestion
        hint.setText(MessageFormat.format(Messages.TooltipHintPressAlt, 
    Platform.OS_MACOSX.equals(Platform.getOS()) ? "Option" : "Alt"));
```

**Review by Nirus2000 — CHANGES_REQUESTED** [review]

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/TimelineChartToolTip.java (hunk 2 above)
Comment (mierin12) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/TimelineChartToolTip.java]: Done with https://github.com/portfolio-performance/portfolio/pull/4359/commits/2b6be85a0c72f3db5cf35e2d2818a0803e131cda

**Review by Nirus2000 — APPROVED** [review]

**Pull request merged**